src/sonic-config-engine/tests/test_cfggen.py

```diff
@@ -647,3 +648,38 @@ def test_show_run_interfaces(self):
         argument = '-a \'{"key1":"value"}\' --var-json INTERFACE'
         output = self.run_script(argument)
         self.assertEqual(output, '')
+
+    def test_minigraph_voq_metadata(self):
+        argument = "-m {} --var-json DEVICE_METADATA".format(self.sample_graph_voq)
+        output = json.loads(self.run_script(argument))
+        self.assertEqual(output['localhost']['switch_id'], '0')
+        self.assertEqual(output['localhost']['switch_type'], 'voq')
+        self.assertEqual(output['localhost']['max_cores'], '16')
```

Comment:
Are we checking for the "asic_id", "asic_name", and "hostname" in any other test case? If not would you please add verification (assertEqual) for ['localhost']['asic_id'], ['localhost']['hostname'] and ['localhost']['asic_name'] here? These are mandatory attributes if "switch_type" is "voq".

Reply:
`asic_id` and `asic_name` are multi-asic attributes, not specific to VoQ switches. I don't think we need either of them on single-asic if `switch_type` is `"voq"`. On single-asic devices where `asic_name` is not specified the system port key is "hostname|portname".

For checks:
- `asic_name` is tested in test_multinpu_cfggen.py but not `asic_id`.
- `hostname` is not specific to VoQ switches but I didn't see any check for this one.

I'll update test_cfggen.py to add a check for `hostname` and `test_multinpu_cfggen.py` for `asic_id`.